app.rb

```diff
@@ -32,11 +32,19 @@ def list_all_people
     puts '-------List of people are as follows:---------'
     @people.each_with_index do |person, index|
       if person.is_a? Student
-        puts "[#{person.class}] - #{index}) - Name: #{person.name},
-        Age: #{person.age}, ID: #{person.id}, Parent Permission: #{person.parent_permission}"
+        name = person.name
+        age = person.age
+        id = person.id
+        par_perm = person.parent_permission
+        grp = person.class
+        puts "[#{grp}] - #{index}) - Name: #{name}, Age: #{age}, ID: #{id}, Parent Permission: #{par_perm}"
       elsif person.is_a? Teacher
-        puts "[#{person.class}] - #{index}) - Name: #{person.name}, Age: #{person.age},
-        ID: #{person.id}, Specialization: #{person.specialization}"
+        grp = person.class
+        name = person.name
+        age = person.age
+        id = person.id
+        spec = person.specialization
+        puts "[#{grp}] - #{index}) - Name: #{name}, Age: #{age}, ID: #{id}, Specialization: #{spec}"
       end
     end
```

Comment:
- [ ] Good job so far, however, we can further improve upon our _create_ and _list_rental_ method in the `# app.rb`, which at present do not adhere to the single responsibility and the YAGNI rule. Let's ensure that each _create_, or _list_ method simply focuses on doing a single/particular task and should contain only objects or instances needed for a given logic of that method. 

  Other logics such as requests for input _(`get.chomps`)_ or _input validations\print statements_ should be placed in the `main.rb`, separate classes or methods. Let's therefore refactor the below methods in the `app.rb` accordingly:
  * `create_person` 
  * `create_book`,
  * `create_rental` and
  * `list_all_rentals_for_a_given_person_id` 

  The example below are methods that have focused responsibility:
```ruby
    class App
  
     . . .
  
      def list_books
        @books.each do |book|
          puts book
        end
      end

      def list_people
        @people.each do |person|
          puts person
       end

      . . .

    end
```

I urge you to read further on the same via [this link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

Reply:
Thank you so much @rloterh for your input.